### PR TITLE
Address DX feedback for data views #8497

### PR DESCRIPTION
**Pull request by mshabarov**
Rename methods name, update javadoc, add extra filtering and sorting methods
Part of #8497

### CODE REVIEW

File: flow-data/src/main/java/com/vaadin/flow/data/provider/AbstractListDataView.java [hunk 1]
```diff
@@ -54,33 +56,19 @@ public AbstractListDataView(
     }
 
     @Override
-    public boolean hasNextItem(T item) {
+    public Optional<T> getNextItem(T item) {
         int index = getItemIndex(item);
         if (index < 0)
-            return false;
-        return getAllItems().skip(index + 1).findAny().isPresent();
+            return Optional.empty();
```
Comment (caalador) [inline flow-data/src/main/java/com/vaadin/flow/data/provider/AbstractListDataView.java:62]: Missing Braces `{}` for if

File: flow-data/src/main/java/com/vaadin/flow/data/provider/AbstractListDataView.java [hunk 2]
```diff
@@ -54,33 +56,19 @@ public AbstractListDataView(
...
-    }
-
-    @Override
-    public T getPreviousItem(T item) {
+    public Optional<T> getPreviousItem(T item) {
         int index = getItemIndex(item);
         if (index <= 0)
-            return null;
-        return getAllItems().skip(index - 1).findFirst().orElse(null);
+            return Optional.empty();
```
Comment (caalador) [inline flow-data/src/main/java/com/vaadin/flow/data/provider/AbstractListDataView.java:70]: Missing Braces `{}` for if

File: flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java [hunk 3]
```diff
@@ -29,66 +32,61 @@
...
      *         item to get next for
-     * @return next item if available, else null
-     */
-    T getNextItem(T item);
-
-    /**
-     * Check if the given item has a previous item in the filtered and sorted
-     * data.
+     * @return next item if available, else empty optional if item
+     *         doesn't exist or filtered out
```
Comment (caalador) [inline flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java:43]: `or not in current filtered items` sounds more precise

File: flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java [hunk 4]
```diff
@@ -29,66 +32,61 @@
...
      */
-    boolean hasPreviousItem(T item);
+    Optional<T> getNextItem(T item);
 
     /**
-     * Get the item before given item from the filtered and sorted data.
+     * Gets the item before given item from the filtered and sorted data.
+     * <p>
+     * Note! Item might be present in the data set, but be filtered out
+     * so that the previous item won't be available.
```
Comment (caalador) [inline flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java:53]: `be filtered out or be the first item so ..`

File: flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java [hunk 5]
```diff
@@ -29,66 +32,61 @@
...
-     * @return true if the item is present and it is not the last item
-     */
-    boolean hasNextItem(T item);
-
-    /**
-     * Get the item after given item from the filtered and sorted data.
+     * Gets the item after given item from the filtered and sorted data.
+     * <p>
+     * Note! Item might be present in the data set, but be filtered out
+     * so that the next item won't be available.
```
Comment (caalador) [inline flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java:38]: `be filtered out or be the last item so ..`

File: flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java [hunk 6]
```diff
@@ -118,45 +117,162 @@
...
      *         if item doesn't exist or collection is not a list
+     * @see #addItemAfter(Object, Object)
      */
     V addItemBefore(T item, T before);
 
     /**
-     * Adds a filter to be applied to all queries. The filter will be used in
-     * addition to any filter that has been set or added previously.
+     * Sets a filter to be applied to the data. Given filter replaces any
+     * previous filter. Setting {@code null} clears filtering.
```
Comment (caalador) [inline flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java:126]: `Given filter replaces any previous filter. Setting {@code null} clears filtering.` -> `The filter replaces any filter that has been set or added previously. {@code null} will clear all filters.`

File: flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java [hunk 7]
```diff
@@ -118,45 +117,162 @@
...
+     *
+     * @see #addFilter(SerializablePredicate)
+     * @see #setFilter(SerializablePredicate)
      */
-    V withFilter(SerializablePredicate<T> filter);
+    V removeFilters();
 
     /**
-     * Sets the comparator to use as the default sorting for data.
+     * Sets the comparator to use as the data default sorting.
```
Comment (caalador) [inline flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java:173]: Drop `data` as it just confuses the line. Also on the next line I would drop `of the data`

File: flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java [hunk 8]
```diff
@@ -118,45 +117,162 @@
...
+     * @param sortComparator
+     *         a comparator to add, not <code>null</code>
+     * @return ListDataView instance
+     *
+     * @see #setSortComparator(SerializableComparator)
+     */
+    V addSortComparator(SerializableComparator<T> sortComparator);
+
+    /**
+     * Removes any data default sorting that has been set or added previously.
```
Comment (caalador) [inline flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java:214]: Remove `data` and only use `default sorting`

File: flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java [hunk 9]
```diff
@@ -118,45 +117,162 @@
...
+     *
+     * @return ListDataView instance
+     *
+     * @see #setSortComparator(SerializableComparator)
+     * @see #addSortComparator(SerializableComparator)
+     */
+    V removeSorting();
+
+    /**
+     * Sets the property and direction to use as the default sorting for data.
```
Comment (caalador) [inline flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java:227]: drop `for data`  also the mention of `data provider` in the first paragraph should be unified through out the sorting JavaDocs

File: flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java [hunk 10]
```diff
@@ -118,45 +117,162 @@
...
+     *
+     * @return ListDataView instance
+     *
+     * @see #addSortOrder(ValueProvider, SortDirection)
+     */
+    <VALUE extends Comparable<? super VALUE>> V setSortOrder(
+            ValueProvider<T, VALUE> valueProvider, SortDirection sortDirection);
+
+    /**
+     * Adds a property and direction to the default sorting for data.
```
Comment (caalador) [inline flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java:253]: Drop `for data`

File: flow-data/src/main/java/com/vaadin/flow/data/provider/AbstractListDataView.java (hunk 1 above)
Comment (mshabarov) [reply flow-data/src/main/java/com/vaadin/flow/data/provider/AbstractListDataView.java]: Done

File: flow-data/src/main/java/com/vaadin/flow/data/provider/AbstractListDataView.java (hunk 2 above)
Comment (mshabarov) [reply flow-data/src/main/java/com/vaadin/flow/data/provider/AbstractListDataView.java]: Done

File: flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java (hunk 3 above)
Comment (mshabarov) [reply flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java]: Done

File: flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java (hunk 4 above)
Comment (mshabarov) [reply flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java]: Done

File: flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java (hunk 5 above)
Comment (mshabarov) [reply flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java]: Done

File: flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java (hunk 6 above)
Comment (mshabarov) [reply flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java]: Done

File: flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java (hunk 7 above)
Comment (mshabarov) [reply flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java]: Done

File: flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java (hunk 8 above)
Comment (mshabarov) [reply flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java]: Done

File: flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java (hunk 9 above)
Comment (mshabarov) [reply flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java]: Done

File: flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java (hunk 10 above)
Comment (mshabarov) [reply flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java]: Done

File: flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java [hunk 11]
```diff
@@ -118,45 +117,164 @@
...
+     * <p>
+     * A filter bound to data set, not to the component. That means this filter
+     * won't be retained when a new data or {@link DataProvider} is set to the
+     * component. Any other component using the same {@link DataProvider} object
+     * would be affected by setting a filter through data view of another
+     * component.
      *
      * @param filter
-     *         the filter to add, not <code>null</code>
+     *         filter to add for the data
```
Comment (caalador) [inline flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java:135]: This is set filter, but the param documents it as `to add`

**Pull request merged**